custom_components/frigate/views.py

```diff
@@ -216,13 +216,26 @@ def _create_path(self, path: str, **kwargs: Any) -> str | None:
 
         camera = path.split("/")[0]
         if path.endswith("clip.mp4"):
-            return f"clips/{camera}-{event_id}.mp4"
+            return f"api/events/{event_id}/clip.mp4"
         return None
 
     def _permit_request(self, request: web.Request, config_entry: ConfigEntry) -> bool:
         """Determine whether to permit a request."""
         return bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
 
+class VodProxyView(ProxyView):
+    """A proxy for clips."""
+
+    url = "/api/frigate/{frigate_instance_id:.+}/vod/{path:.*}"
+    extra_urls = ["/api/frigate/vod/{path:.*}"]
+
+    name = "api:frigate:vod"
+    requires_auth = False
```

Comment:
Can we keep auth, given the existing clips proxy uses auth?

Reply:
The way HLS works, Media Browser will send the `authSig` query string parameter on the initial `index.m3u8` request. but it then gets processed and additional reaquests will be made for `segment-n.ts`, which will not have the `authSig` parameter, and thus fail this proxy.

I could potentially use https://github.com/kaltura/nginx-secure-token-module to pass `authSig` down to the generated segments, but need to do additional testing to see if that is feasible.

Reply:
This has been resolved, although it's a little weird. Home Assistant's URL signing policy only signs a single URL. In the case of HLS, that is the manifest file. After HLS loads the manifest, it proxies each segment through as well. I made a change to pass that authSig down to the segments, but then the request path doesn't match. What I ended up doing is disabling the built-in auth, and only validate that the parent path of the signature matches for each of the segments instead of the full path. That way people still can't directly enter a segment URL.